### PR TITLE
mm: fix the user L1 mmu entries caculation

### PR DESCRIPTION
**Pull request by prime-zeng**
Signed-off-by: Zeng Tao <prime.zeng@hisilicon.com>

### CODE REVIEW

File: core/arch/arm/include/mm/tee_mmu_defs.h [hunk 1]
```diff
@@ -32,7 +32,7 @@
 #define TEE_MMU_TTBCR_N_VALUE		7
 
 /* Number of sections in ttbr0 when user mapping activated */
-#define TEE_MMU_UL1_NUM_ENTRIES         ((1 << TEE_MMU_TTBCR_N_VALUE) / 4)
+#define TEE_MMU_UL1_NUM_ENTRIES         (1 << (12 - TEE_MMU_TTBCR_N_VALUE))
```
Comment (jenswi-linaro) [inline core/arch/arm/include/mm/tee_mmu_defs.h]: This calculation is obviously not trivial, please add a comment explaining what's going on.

Comment (prime-zeng) [inline core/arch/arm/include/mm/tee_mmu_defs.h]: ok

Comment (prime-zeng) [inline core/arch/arm/include/mm/tee_mmu_defs.h]: @jenswi-linaro   I think we can just modity the TEE_MMU_TTBCR_N_VALUE to change the user TA address space, is that right?

**Comment by jenswi-linaro** [comment]
`Reviewed-by: Jens Wiklander <jens.wiklander@linaro.org>`

**Comment by jbech-linaro** [comment]
Please update the commit message, there is a spelling error:
`s/caculation/calculation/`

Other than that 
`Reviewed-by: Joakim Bech <joakim.bech@linaro.org>`

File: core/arch/arm/include/mm/tee_mmu_defs.h (hunk 1 above)
Comment (jenswi-linaro) [inline core/arch/arm/include/mm/tee_mmu_defs.h]: That should be possible, but it has never been tested and may break something.

**Comment by prime-zeng** [comment]
Applied

File: core/arch/arm/include/mm/tee_mmu_defs.h (hunk 1 above)
Comment (jenswi-linaro) [inline core/arch/arm/include/mm/tee_mmu_defs.h]: Just change it and run xtest. On the other hand, why would you like to change it? Do you need a larger VM space for the user TAs or what?

**Comment by jforissier** [comment]
Rebased and merged.

File: core/arch/arm/include/mm/tee_mmu_defs.h (hunk 1 above)
Comment (prime-zeng) [inline core/arch/arm/include/mm/tee_mmu_defs.h]: Ok,  i will test it , some DRM TAs need to deal with the video frame,  i need to map the video buffer into the TA VM space, 32M is not enough for 4K resolution videos.

**Pull request closed without merging**